### PR TITLE
specify Xcode version in Jenkins jobs

### PR DESCRIPTION
**Pull request by jpsim**
Addresses #4620

### CODE REVIEW

**Comment by jpsim** [comment]
This will need to be amended once we want to resume running jobs with non-default Swift toolchains again (such as when Xcode 9 comes out with a Swift 3 compilation mode) but for now this should be sufficient.

**Comment by tgoyne** [comment]
Do the release scripts need to be updated as well so that we don't accidentally cut a release using the 8.3 beta?

**Comment by jpsim** [comment]
You're right @tgoyne, done in ee64c1f.

File: scripts/swift-version.sh [hunk 1]
```diff
@@ -3,6 +3,42 @@ get_swift_version() {
...
 }
 
+get_xcode_version() {
+    xcodebuild_command=$@
+    $xcodebuild_command -version 2>/dev/null | sed -ne 's/^Xcode \([^\b ]*\).*/\1/p'
+}
+
+find_xcode_with_version() {
+    # First check if the currently active one is fine
+    version="$(get_xcode_version xcodebuild)"
```
Comment (larkost) [inline scripts/swift-version.sh]: The quotes here are superfluous, but not harmful.

File: scripts/swift-version.sh [hunk 2]
```diff
@@ -3,6 +3,42 @@ get_swift_version() {
...
+    if [[ "$version" = "$1" ]]; then
+        export DEVELOPER_DIR="$(xcode-select -p)"
+        export REALM_SWIFT_VERSION=$(get_swift_version xcrun swift)
+        return 0
+    fi
+
+    local xcodes dev_dir version
```
Comment (larkost) [inline scripts/swift-version.sh]: This should probably move up before line 13.

File: scripts/swift-version.sh [hunk 3]
```diff
@@ -34,7 +70,9 @@ find_xcode_for_swift() {
     return 1
 }
 
-if [[ "$REALM_SWIFT_VERSION" ]]; then
+if [[ "$REALM_XCODE_VERSION" ]]; then
```
Comment (larkost) [inline scripts/swift-version.sh]: I would generally use `-n` to be certain, but I think that this will work in any event.

**Pull request merged**